src/mscorlib/src/System/Threading/Mutex.cs

```diff
@@ -2,218 +2,123 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-//
-/*=============================================================================
-**
-**
-**
-** Purpose: synchronization primitive that can also be used for interprocess synchronization
-**
-**
-=============================================================================*/
+
+using System.IO;
+using Microsoft.Win32;
+using Microsoft.Win32.SafeHandles;
+using System.Runtime.InteropServices;
+using System.Diagnostics.Contracts;
+using System.Diagnostics;
 
 namespace System.Threading
 {
-    using System;
-    using System.Threading;
-    using System.Runtime.CompilerServices;
-    using System.IO;
-    using Microsoft.Win32;
-    using Microsoft.Win32.SafeHandles;
-    using System.Runtime.InteropServices;
-    using System.Runtime.Versioning;
-    using System.Security;
-    using System.Diagnostics;
-    using System.Diagnostics.Contracts;
-
+    /// <summary>
+    /// Synchronization primitive that can also be used for interprocess synchronization
+    /// </summary>
     public sealed class Mutex : WaitHandle
     {
         private const uint AccessRights =
             (uint)Win32Native.MAXIMUM_ALLOWED | Win32Native.SYNCHRONIZE | Win32Native.MUTEX_MODIFY_STATE;
 
-        private static bool dummyBool;
-
-        internal class MutexSecurity
+        public Mutex(bool initiallyOwned, string name, out bool createdNew)
         {
-        }
+            VerifyNameForCreate(name);
+            Contract.EndContractBlock();
 
-        public Mutex(bool initiallyOwned, String name, out bool createdNew)
-            : this(initiallyOwned, name, out createdNew, (MutexSecurity)null)
-        {
+            CreateMutexCore(initiallyOwned, name, out createdNew);
         }
 
-        internal unsafe Mutex(bool initiallyOwned, String name, out bool createdNew, MutexSecurity mutexSecurity)
+        public Mutex(bool initiallyOwned, string name)
         {
-            if (name == string.Empty)
-            {
-                // Empty name is treated as an unnamed mutex. Set to null, and we will check for null from now on.
-                name = null;
-            }
-#if PLATFORM_WINDOWS
-            if (name != null && System.IO.Path.MaxPath < name.Length)
-            {
-                throw new ArgumentException(SR.Format(SR.Argument_WaitHandleNameTooLong, Path.MaxPath), nameof(name));
-            }
-#endif // PLATFORM_WINDOWS
+            VerifyNameForCreate(name);
```

Comment:
This used to be under Windows ifdef, but it is enabled for all platforms now

Reply:
On corert it's throwing when a name is provided: https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/src/System/Threading/Mutex.Unix.cs#L16

Should this be the same here? I'm not sure about this...

Reply:
Expected - CoreRT does not support named mutexes on Unix today.